src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -788,6 +802,29 @@ public FormValidation doCheckMaxRequestsPerHostStr(@QueryParameter String value)
             }
         }
 
+        public FormValidation doCheckDirectConnection(@QueryParameter boolean value, @QueryParameter String jenkinsUrl) throws IOException, ServletException {
+            if(value) {
+                if(!isEmpty(jenkinsUrl)) return FormValidation.warning("no need to configure Jenkins URL if direct connection is enabled");
+            } else {
+                if(isEmpty(jenkinsUrl)) return FormValidation.warning("configure either Direct Connection or Jenkins URL ");
+            }
+            return FormValidation.ok();
+        }
+
+        public FormValidation doCheckJenkinsUrl(@QueryParameter String value, @QueryParameter boolean directConnection) throws IOException, ServletException {
```

Comment:
You do not need both of these. Probably you want to validate whatever field appears lower in the form, so that warnings are visible to anyone changing either one. Or keep `doCheckJenkinsUrl` limited to checking URL syntax, and leave the interaction to `doCheckDirectConnection`.

Reply:
done